modules/dataframe/src/properties/dataframecolormapproperty.cpp

```diff
@@ -0,0 +1,123 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/dataframe/properties/dataframecolormapproperty.h>
+
+namespace inviwo {
+
+const std::string DataFrameColormapProperty::classIdentifier = "org.inviwo.DataFrameColormapProperty";
+std::string DataFrameColormapProperty::getClassIdentifier() const { return classIdentifier; }
+
+DataFrameColormapProperty::DataFrameColormapProperty(std::string identifier,
+                                                     std::string displayName,
+                                                     DataInport<DataFrame>& port)
+    : CompositeProperty(identifier, displayName)
+    , selectedColorAxis{"selectedColorAxis", "Column", port, false, 1}
+    , overrideColormap("overrideColormap", "Override", false)
+    , tf{"tf", "Colormap"}
+    , dataframe_(port.getData()) {
+
+    overrideColormap.onChange([&]() {
+        if (*overrideColormap) {
+            colormapChanged_.reset();
+            for (auto p : colormaps_) {
+                p->setVisible(false);
+            }
+        } else {
+            selectedColorAxis.propertyModified();
+        }
+        tf.setReadOnly(!*overrideColormap);
+    });
+
+
+    selectedColorAxis.onChange([&]() {
+        if (colormaps_.size() <= selectedColorAxis || overrideColormap) return;
+
+        for (auto i = 0; i < colormaps_.size(); i++) {
+            colormaps_[i]->setVisible(i == selectedColorAxis);
+        }
+
+        auto updateColormap = [&]() {
+            if (colormaps_.size() <= selectedColorAxis) return;
+            auto cm = dynamic_cast<const ColormapProperty*>(colormaps_[selectedColorAxis]);
+            if (cm) *tf = cm->getTransferFunction();
+        };
+        colormapChanged_ = dynamic_cast<ColormapProperty*>(colormaps_[selectedColorAxis])
+                               ->onChangeScoped(updateColormap);
+        updateColormap();
+    });
+
+    addProperties(selectedColorAxis, overrideColormap, tf);
+
+    port.onChange([this, portPtr = &port]() {
+        if (portPtr->hasData()) {
+            createOrUpdateProperties(portPtr->getData());
+        }
+    });
+}
+
+void DataFrameColormapProperty::createOrUpdateProperties(
+    std::shared_ptr<const DataFrame> dataframe) {
+    if (!dataframe || dataframe->getNumberOfColumns() <= 1) return;
+    dataframe_ = dataframe;
+    auto oldColormapProperties = colormaps_;
+    colormaps_.clear();
+    for (size_t i = 0; i < dataframe_->getNumberOfColumns(); i++) {
+        auto c = dataframe_->getColumn(i);
+        std::string displayName = "Settings";
+        std::string identifier = util::stripIdentifier(c->getHeader());
+
+        auto colormapProp = [&]() -> ColormapProperty* {
+            if (auto p = getPropertyByIdentifier(identifier)) {
```

Comment:
why not remove all properties stored in `colormaps_` beforehand?

Reply:
you are not replacing them anyway.

Reply:
They are removed on line 89 and reused if the same column in the new DataFrame is found